001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+            discarded.
+
+          * Otherwise the DWARF expression is ill-formed.
+
+            > [For further discussion...]
+            > Could define this case as returning an implicit location
+            > description as if the DW_OP_implicit operation is
+            > performed.
+
+      * If the current result kind specifies a value, then:
+
+          * If the top stack entry is a value, or can be converted to
+            one (see 2.5.4.4.3 Memory Location Description Operations),
+            then the result is that, possibly converted, value. Any
+            other entries on the stack are discarded.
+
+          * Otherwise the DWARF expression is ill-formed.
+
+      * If the current result kind is not specified, then:
+
+          * If the stack is empty, the result is a location description
+            with one undefined location description.
+
+            [non-normative] This rule is for backwards compatibility
+            with DWARF Version 5 which uses an empty operation
+            expression for this purpose.
+
+            > [For further discussion...]
+            > This rule is consistent with the rule above for when a
+            > location description is requested. However, GDB appears to
+            > report this as an error and no GDB tests appear to cause
+            > an empty stack for this case.
+
+          * Otherwise, the top stack entry is returned. Any other
+            entries on the stack are discarded.
+
+    An operation expression is encoded as a byte block with some form of
+    prefix that specifies the byte count. It can be used:
+
+      * as the value of a debugging information entry attribute that is
+        encoded using class exprloc (see 7.5.5 Classes and Forms).
+
+      * as the operand to certain operation expression operations.
+
+      * as the operand to certain call frame information operations (see
+        6.4 Call Frame Information).
+
+      * in location list entries (see 2.5.x DWARF Location List
+        Expressions).
+    --------------------------------------------------------------------
+
+In Section 2.5.1.1 Literal Encodings, replace the description of
+DW_OP_const_type with the following:
+
+    --------------------------------------------------------------------
+    DW_OP_const_type has three operands. The first is an unsigned LEB128
+    integer DR that represents the byte offset of a debugging
+    information entry D relative to the beginning of the current
+    compilation unit, that provides the type T of the constant value.
+    The second is a 1-byte unsigned integral constant S. The third is a
+    block of bytes B, with a length equal to S.
+
+    TS is the bit size of the type T. The least significant TS bits of B
+    are interpreted as a value V of the type D. It pushes the value V
+    with the type D.
+
+    The DWARF is ill-formed if D is not a DW_TAG_base_type debugging
+    information entry in the current compilation unit, or if TS divided
```

Comment:
Also abbreviate 'current compilation unit'?

Reply:
Where would the definition for the abbreviation belong? Should DWARF have a glossary? I did not abbreviate this term as it did not seem to be used that much.

Reply:
What I meant was to give a name to the `current compilation unit` in line 365f, e.g. `CU`, so we can refer to `CU` here.

Reply:
Looks like section "6.1.1.2 Structure of the Name Index" defines CU in two places:

"A list of compile units (CUs) referenced by this index."

"Compilation Unit (CU), a reference to an entry in the list of CUs. "

I also see "CU" being used earlier in the spec with no definition. But I also see many uses of "compilation unit" without an abbreviation. I think deciding this is probably a spec editor decision.

Reply:
What I meant was that the text gives a name to `a debug information entry D` and refers to `D` for that entry later on.  For the `current compilation unit` it does not and instead repeats the term later on.  Could be that's because the current compilation unit is part of the evaluation context.

Reply:
Are, I did not notice that. I think it is likely the reason you mention.